### PR TITLE
Tests and cleanup

### PR DESCRIPTION
**Pull request by scymtym**
* Tests for #8 
* ASDF cleanup
* Whitespace and typos

### CODE REVIEW

File: tests/test-regressions.lisp [hunk 1]
```diff
@@ -5,3 +5,34 @@
...
+                    (with-simple-restart (abort "Abort")
+                      (handler-bind
+                          ((error (lambda (condition)
+                                    (log:error "caught error: ~A" condition)
+                                    (abort))))
+                        (log:warn "~@{~A ~A~}" 1)))))))
+    (is (search (format nil "WARN - 1 ~@
+                             ERROR - caught error: ")
+                output))))
+
```
Comment (dkochmanski) [inline tests/test-regressions.lisp]: This test doesn't look right (with regard to `is` assertion), locally I see
```
" <WARN> [14:08:41] log4cl fileDssrBB (xxx)
<ERROR> [14:08:41] log4cl fileDssrBB (xxx)
* caught error: 
Missing argument
\"~@{~A ~A~}\"
        ^
"
```

and this won't match what is covered in `format`. Same goes for the test below. I think that we should simply assert, that the code finishes (without causing error). Do I miss something?

**Comment by dkochmanski** [comment]
ah, I see, it is specially configured for tests. either way, shouldn't we just test, that locking attempt doesn't cause the deadlock in these situations? that will remove ccl failure as well (which has nothing to do with logging).

File: tests/test-regressions.lisp (hunk 1 above)
Comment (dkochmanski) [reply tests/test-regressions.lisp]: ah, I see, it is specially configured for tests. either way, shouldn't we just test, that locking attempt doesn't cause the deadlock in these situations? that will remove ccl failure as well (which has nothing to do with logging).

**Comment by scymtym** [comment]
I can see your point and that's what I did at first (i.e. no checks besides ``finishes``). I added the more elaborate checks because I worry that a change in behavior that still satisfies ``finishes`` could go unnoticed otherwise.

If you think this is too much trouble/too brittle w.r.t. implementation quirks, I can go back to just ``finishes``.

**Comment by dkochmanski** [comment]
I would go with finishes then, so we have failure only when #8 has indeed regression about locks.

**Comment by scymtym** [comment]
OK.

**Comment by dkochmanski** [comment]
looks good, thanks!

**Pull request merged**